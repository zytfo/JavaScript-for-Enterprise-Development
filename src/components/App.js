import React, { Component } from "react";

import '../styles/App.css';

class App extends Component { render() {
        return (
            <div>
                <h1>This is my Steam Inventory Explorer application.</h1>
            </div>
        );
    }
}

export default App;