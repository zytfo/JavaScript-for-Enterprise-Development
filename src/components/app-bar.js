import style from "../styles/app-bar.css"
import React from "react"
import Typography from "@material-ui/core/Typography"
import Toolbar from "@material-ui/core/Toolbar"
import AppBar from "@material-ui/core/AppBar"
import { withRouter } from "react-router-dom"
import { connect } from 'react-redux'


class ItemsAppBar extends React.Component {
    render() {
        return (
            <AppBar position="static">
                <Toolbar className={style.toolbar}>
                    <Typography variant="h6" color="inherit" noWrap className={style.title}>
                        Steam Inventory Explorer
                    </Typography>
                </Toolbar>
            </AppBar>
        )
    }
}

export default connect()(withRouter(ItemsAppBar))