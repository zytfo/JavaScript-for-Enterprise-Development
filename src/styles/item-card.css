.card {
    margin-top: 50px;
}

.media {
    height: 250px;
    width: 350px;
}