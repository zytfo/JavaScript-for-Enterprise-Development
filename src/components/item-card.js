import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import React from 'react';
import styles from '../styles/item-card.css';

export const ItemCard = ({ item }) => (
    <Grid item sm={'6'} md={'4'} lg={'3'}>
        <Card className={styles.card}>
            <CardMedia
                className={styles.media}
                image={"https://steamcommunity-a.akamaihd.net/economy/image/class/440/".concat(item.classid).concat("/333fx171f.jpg")}
                title="Name"
            />
            <CardContent>
                <Typography gutterBottom variant="h5" component="h2">
                    {item.name}
                </Typography>
                <Typography>
                    Type: {item.type}
                </Typography>
            </CardContent>
            {/* <CardActions>
                <Button size="small" color="primary">
                    <span className={'class'}>Details</span>
                </Button>
            </CardActions> */}
        </Card>
    </Grid>
)