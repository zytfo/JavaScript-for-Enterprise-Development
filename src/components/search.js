import React from 'react';
import styles from '../styles/search.css';
import InputBase from "@material-ui/core/InputBase";
import Paper from "@material-ui/core/Paper";
import Divider from "@material-ui/core/Divider";
import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';
import Typography from "@material-ui/core/Typography";
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import MoreVertIcon from '@material-ui/icons/MoreVert';

const options = [
    'Team Fortress 2',
    'CS: GO',
    'Dota 2'
];

const ITEM_HEIGHT = 48;
let gameId = 440;

export class SearchBar extends React.Component {
    state = {
        id: "",
        anchorEl: null,
        game: ""
    };

    buildDetailsClickHandler = () => {
        // this.props.history.push(`/profile/${this.state.id}/${this.state.game}/items/`);
        this.props.history.push(`/profile/${this.state.id}/items/`);
    };

    updateInput = (event) => {
        this.setState({
            id: event.target.value
        })
    };

    handleClick = event => {
        this.setState({ anchorEl: event.currentTarget });
    };

    handleClose = (event) => {
        switch (event.nativeEvent.target.outerText) {
            case 'Team Fortress 2':
                gameId = 440;
                break;
            case 'CS: GO':
                gameId = 730;
                break;
            case 'Dota 2':
                gameId = 570;
                break;
        }
        this.setState( {
            anchorEl: null,
            game: gameId
        });
        console.log(this.state.game);
    };

    render() {
        const { anchorEl } = this.state;
        const open = Boolean(anchorEl);

        return (
            <div>
                <Typography className={styles.row} gutterBottom variant="h5" component="h2">
                    Enter a valid Steam id (e.g. 76561198205886600). Otherwise it will never load.
                </Typography>
                <Paper className={styles.root} elevation={1}>
                    <InputBase className={styles.input} value={this.state.id} placeholder="Search Steam Profile ID" onChange={this.updateInput} onKeyPress={event => {
                        if (event.key === 'Enter') {
                            this.buildDetailsClickHandler();
                        }
                    }}/>
                    <IconButton
                        aria-label="More"
                        aria-owns={open ? 'long-menu' : undefined}
                        aria-haspopup="true"
                        onClick={this.handleClick}
                    >
                        <MoreVertIcon />
                    </IconButton>
                    <Menu
                        id="long-menu"
                        anchorEl={anchorEl}
                        open={open}
                        onClose={this.handleClose}
                        PaperProps={{
                            style: {
                                maxHeight: ITEM_HEIGHT * 4.5,
                                width: 200,
                            },
                        }}
                    >
                        {options.map(option => (
                            <MenuItem key={option} selected={option === this.state.game} onClick={this.handleClose}>
                                {option}
                            </MenuItem>
                        ))}
                    </Menu>
                    <IconButton className={styles.iconButton} aria-label="Search" onClick={this.buildDetailsClickHandler}>
                        <SearchIcon />
                    </IconButton>
                    <Divider className={styles.divider} />
                </Paper>
            </div>
        );
    }
}