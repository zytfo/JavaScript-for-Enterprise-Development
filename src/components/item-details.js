import React from 'react'
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Grid from '@material-ui/core/Grid';

import styles from "../styles/item-details.css"
import ReactLoading from 'react-loading';
import stylesLoading from '../styles/loading-screen.css';
import {ItemCard} from "./item-card";

export class ItemDetails extends React.Component {

    state = { item: void 0 };

    componentDidMount() {
        if (this.props.match.params) {
            this.setState({
                item: this.props.location.state,
                classid: this.props.match.params.itemid,
                gameid: this.props.match.params.gameid
            })
        }
    }


    render() {
        const { item } = this.state;
        if (!item) {
            return <div><ReactLoading className={stylesLoading.loading} type={"spokes"} color={"#1c2735"} height={'10%'} width={'10%'}/></div>
        }

        return (
            <Grid container justify="center" alignItems="center">
                <Grid item xs={12} md={6} lg={4}>
                    <Card className={styles.card}>
                        <CardMedia
                            className={styles.media}
                            image={`https://steamcommunity-a.akamaihd.net/economy/image/class/${this.state.gameid}/${this.state.classid}/333fx171f.jpg`}
                            title="Avatar"
                        />
                        <CardContent>
                            <Typography gutterBottom variant="h5" component="h2">
                                {item.name}
                            </Typography>
                            <Typography>
                                {item.type}
                            </Typography>
                            <hr/>
                            <Table className={''}>
                                <TableBody>
                                    <TableRow>
                                        <TableCell component="th" scope="row">Name color</TableCell>
                                        <TableCell align="right">{item.name_color}</TableCell>
                                    </TableRow>

                                    <TableRow>
                                        <TableCell component="th" scope="row">Tradability</TableCell>
                                        <TableCell align="right">{item.tradable}</TableCell>
                                    </TableRow>

                                    <TableRow>
                                        <TableCell component="th" scope="row">Marketable</TableCell>
                                        <TableCell align="right">{item.marketable}</TableCell>
                                    </TableRow>

                                    <TableRow>
                                        <TableCell component="th" scope="row">Descriptions</TableCell>
                                    </TableRow>

                                    {Object.entries(item.descriptions).map(description =>
                                        <TableRow
                                                   key={description[0]}
                                                   item={description[1]}>
                                            <TableCell component="th" scope="row"></TableCell>
                                            <TableCell align="right">
                                                {description[1].value}
                                            </TableCell>
                                        </TableRow>
                                    )}
                                </TableBody>
                            </Table>
                        </CardContent>
                    </Card>
                </Grid>
            </Grid>
        )
    }
}
