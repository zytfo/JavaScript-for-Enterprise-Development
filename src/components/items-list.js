import ReactLoading from 'react-loading';
import React from "react";
import { ItemCard } from "./item-card"
import styles from '../styles/loading-screen.css';
import style from '../styles/item-list.css';
import { Grid } from './grid';
import { connect } from "react-redux"
import { loadItemsActionCreator } from "../redux/actionCreators/load-items"

let storedItem;
let currentProps = null;

class ItemsList extends React.Component {
    getId = (props) => props.location.pathname.replace(`/profile/${this.props.match.params.id}/${this.props.match.params.gameid}/items/`, `${this.props.match.params.id}`);
    getGameId = (props) => props.location.pathname.replace(`/profile/${this.props.match.params.id}/${this.props.match.params.gameid}/items/`, `${this.props.match.params.gameid}`);

    componentDidMount() {
        if (currentProps === null) {
            currentProps = this.props;
            this.props.loadItems(this.getId(this.props), this.getGameId(this.props));
        } else if (currentProps.location.pathname !== this.props.location.pathname) {
            currentProps = this.props;
            this.props.loadItems(this.getId(this.props), this.getGameId(this.props));
        }
    }

    buildDetailsClickHandler = (item) => () => {
        storedItem = item;
        this.props.match.params.classid = item.classid;
        this.props.history.push(
            `/profile/${this.props.match.params.id}/${this.props.match.params.gameid}/item/${this.props.match.params.classid}`,
            item
        );
    };

    render() {
        if (!this.props.data) {
            return <div><ReactLoading className={styles.loading} type={"spokes"} color={"#1c2735"} height={'10%'} width={'10%'}/></div>
        }
        if (this.props.loadFailed) {
            return <h3>Error loading data from API</h3>
        }

        return (
            <div className={style.body}>
                <Grid container>
                    {Object.entries(this.props.data).map(item =>
                        <ItemCard
                            key={item[0]}
                            item={item[1]}
                            handleDetailsClick={this.buildDetailsClickHandler(item[1])}
                        />
                    )}
                </Grid>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    data: state.data,
    loadFailed: state.itemsLoadingFailed
});

const mapDispatchToProps = (dispatch) => ({
    loadItems: (id, gameid) => dispatch(loadItemsActionCreator(id, gameid))
});

export default connect(mapStateToProps, mapDispatchToProps)(ItemsList)