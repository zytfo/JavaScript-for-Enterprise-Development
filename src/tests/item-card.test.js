import React from 'react'
import { ItemCard } from '../components/item-card'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import { shallow, mount, render } from 'enzyme';

test('Grid renders as container', () => {
    const shallowComponent = shallow(<ItemCard item={{origin: ''}} handleDetailsClick={() =>{}} />);
    const mountComponent = mount(<ItemCard item={{origin: ''}} handleDetailsClick={() =>{}} />);
    expect(shallowComponent).toBeDefined();
    expect(shallowComponent.debug()).toMatchSnapshot()
});