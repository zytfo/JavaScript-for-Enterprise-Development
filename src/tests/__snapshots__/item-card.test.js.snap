// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders as container 1`] = `
"<Grid item={true} sm={6} md={4} lg={3} direction=\\"row\\">
  <WithStyles(Card) className=\\"card\\">
    <WithStyles(CardMedia) className=\\"media\\" image=\\"https://steamcommunity-a.akamaihd.net/economy/image/class/undefined/undefined/333fx171f.jpg\\" title=\\"Name\\" />
    <WithStyles(CardContent)>
      <WithStyles(Typography) gutterBottom={true} variant=\\"h5\\" component=\\"h2\\" />
      <WithStyles(Typography)>
        Type: 
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
    <WithStyles(CardActions)>
      <WithStyles(Button) size=\\"small\\" color=\\"primary\\" onClick={[Function: handleDetailsClick]}>
        <span className=\\"class\\">
          Details
        </span>
      </WithStyles(Button)>
    </WithStyles(CardActions)>
  </WithStyles(Card)>
</Grid>"
`;
